torch/nn/modules/module.py

```diff
@@ -1563,6 +1586,11 @@ def load(module, prefix=''):
                 if child is not None:
                     load(child, prefix + name + '.')
 
+            # Note that the hook can modify missing_keys and unexpected_keys.
+            incompatible_keys = _IncompatibleKeys(missing_keys, unexpected_keys)
+            for hook in module._load_state_dict_post_hooks.values():
+                hook(module, incompatible_keys)
```

Comment:
Since quite a few of our hooks are expected to return the new value of the argument when we want to modify it, it would be nice to add a check here that the hook returns `None` (hinting that the input must be modified inplace).